src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -148,31 +148,32 @@ export class TerminalPanel extends Panel {
 				// Drop selection and focus terminal on Linux to enable middle button paste when click
 				// occurs on the selection itself.
 				this._terminalService.getActiveInstance().focus();
-			} else if (event.which === 3) {
-				if (this._terminalService.configHelper.getRightClickCopyPaste()) {
-					let terminal = this._terminalService.getActiveInstance();
-					if (terminal.hasSelection()) {
-						terminal.copySelection();
-						terminal.clearSelection();
-					} else {
-						terminal.paste();
-					}
+			}
+		}));
+		this._register(DOM.addDisposableListener(this._parentDomElement, 'contextmenu', (event: MouseEvent) => {
+			if (this._terminalService.configHelper.getRightClickCopyPaste()) {
```

Comment:
Well we don't want to regress with right click copy paste so we'll probably want to keep a boolean member variable that we'll use to "cancel" the contextmenu event.